extmod/machine_mem.h

```diff
@@ -41,10 +41,28 @@ extern const machine_mem_obj_t machine_mem8_obj;
 extern const machine_mem_obj_t machine_mem16_obj;
 extern const machine_mem_obj_t machine_mem32_obj;
 
-// It is expected that a port will provide the following 2 functions.
-// We define the prototypes here, but the modmachine.c file for a port should
-// provide the implementation
-uintptr_t machine_mem_get_read_addr(mp_obj_t addr_o, uint align);
-uintptr_t machine_mem_get_write_addr(mp_obj_t addr_o, uint align);
+// If you wish to override the functions for mapping the machine_mem read/write
+// address, then add a #define for MICROPY_MACHINE_MEM_GET_READ_ADDR and/or
+// MICROPY_MACHINE_MEM_GET_WRITE_ADDR. Since the prototypes are identical, it
+// is allowable for both of the macros to evaluate the to same function.
+// 
+// It is expected that the modmachine.c file for a given port will provide the
+// implementations, if the default implementation isn't used.
+//
+// The default implementation, called machine_mem_get_addr is an identity
+// function and can be found in machine_mem.c
 
-#endif /*  MICROPY_EXTMOD_MACHINE_MEM */
+#if !defined(MICROPY_MACHINE_MEM_GET_READ_ADDR)
```

Comment:
This might have been put here to complement comment above, but doesn't really belong here - this logic can happen in .c, without macro creep (and especially without naming internal macros as if they were public, but again, doing it in .c, will save from need to have such macros at all).

~~~
#ifndef MICROPY_MACHINE_MEM_GET_WRITE_ADDR
int f() {}
#define MICROPY_MACHINE_MEM_GET_WRITE_ADDR foo
#endif
~~~